library/src/androidTest/java/com/google/android/material/motion/runtime/SchedulerTest.java

```diff
@@ -0,0 +1,163 @@
+    transaction = new Transaction();
+  }
+
+  public void testInitialSchedulerState() {
+    assertTrue(scheduler.getState() == Scheduler.IDLE);
+  }
+
+  public void testStandardPlanPerformanceSchedulerState() {
+    transaction.addNamedPlan(new StandardPlan("standard"), "plan", textView);
+    scheduler.commitTransaction(transaction);
+
+    assertTrue(scheduler.getState() == Scheduler.IDLE);
+  }
+
+  public void testManualPlanPerformanceSchedulerState() {
+    transaction.addNamedPlan(new ManualPlan("manual"), "plan", textView);
+    scheduler.commitTransaction(transaction);
+
+    assertTrue(scheduler.getState() == Scheduler.ACTIVE);
+  }
+
+  public void testDelegatePlanPerformanceSchedulerState() {
+    transaction.addNamedPlan(new DelegatedPlan("delegated"), "plan", textView);
+    scheduler.commitTransaction(transaction);
+
+    assertTrue(scheduler.getState() == Scheduler.IDLE);
+  }
+
+  public void testAddingSchedulerListeners() {
+    TestSchedulerListener firstListener = new TestSchedulerListener();
+    TestSchedulerListener secondListener = new TestSchedulerListener();
+    scheduler.addStateListener(firstListener);
+    scheduler.addStateListener(secondListener);
+
+    transaction.addNamedPlan(new ManualPlan("manual"), "plan", textView);
+    scheduler.commitTransaction(transaction);
+
+    assertTrue(firstListener.getState() == Scheduler.ACTIVE);
+    assertTrue(secondListener.getState() == Scheduler.ACTIVE);
+  }
+
+  private static class StandardPlan extends Plan {
+
+    private final String text;
+
+    private StandardPlan(String text) {
+      this.text = text;
+    }
+
+    @Override
+    public Class<? extends Performer> getPerformerClass() {
+      return StandardPlanPerformer.class;
+    }
+  }
+
+  private static class ManualPlan extends Plan {
+
+    private final String text;
+
+    private ManualPlan(String text) {
+      this.text = text;
+    }
+
+    @Override
+    public Class<? extends Performer> getPerformerClass() {
+      return ManualPerformer.class;
+    }
+  }
+
+  private static class DelegatedPlan extends Plan {
+
+    private final String text;
+
+    private DelegatedPlan(String text) {
+      this.text = text;
+    }
+
+    @Override
+    public Class<? extends Performer> getPerformerClass() {
+      return DelegatedPerformer.class;
+    }
+  }
+
+  public static class StandardPlanPerformer extends Performer implements Performer.PlanPerformance {
+
+    @Override
+    public void addPlan(Plan plan) {
+      StandardPlan standardPlan = (StandardPlan) plan;
+      TextView target = getTarget();
+      target.setText(target.getText() + " " + standardPlan.text);
+    }
+  }
+
+  public static class ManualPerformer extends Performer implements Performer.ManualPerformance {
+
+    @Override
+    public int update(float deltaTimeMs) {
+      return Scheduler.ACTIVE;
+    }
+  }
+
+  public static class DelegatedPerformer extends Performer implements Performer.DelegatedPerformance {
```

Comment:
the usual way to implement a DelegatedPerformer is to also implement PlanPerformance.

```
@Override
public void addPlan() {
    view.animate().setListener(new Listener() {
      private Token token;
      
      @Override
      public void onStart() {
        token = callback.onDelegatedPerformanceStart(this);
      }
      
      @Override
      public void onEnd() {
        callback.onDelegatedPerformanceEnd(this, token);
      }
    });
}
```